bot/exts/evergreen/githubinfo.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from datetime import datetime
+
+import aiohttp
+import discord
+from discord.ext import commands
+from discord.ext.commands.cooldowns import BucketType
+
+log = logging.getLogger(__name__)
+
+
+class GithubInfo(commands.Cog):
+    """Fetches GitHub."""
+
+    def __init__(self, bot: commands.Bot):
+        self.bot = bot
+
+    async def fetch_data(self, url: str) -> dict:
+        """Retrieve data as a dictionary."""
+        async with aiohttp.ClientSession() as cs:
+            async with cs.get(url) as r:
+                return await r.json()
+
+    @commands.command(name='github', aliases=['gh'])
+    @commands.cooldown(1, 5, BucketType.user)
+    async def get_github_info(self, ctx: commands.Context, username: str = 'python-discord') -> None:
+        """
+        Fetches a user's GitHub information.
+
+        User defaults to python-discord.
+
+        Usage: .gh [username]
+        """
+        # fetch user data
+        user_data = await self.fetch_data(f"https://api.github.com/users/{username}")
+
+        # user_data will not have a message key if the user exists
+        if user_data.get('message') is not None:
+            return await ctx.send(embed=discord.Embed(title=f"The profile `{username}` was not found", colour=0xff0022))
+
+        # fetching organization data and forming the organization string
+        org_data = await self.fetch_data(user_data['organizations_url'])
+
+        orgs = [f"[{org['login']}](https://github.com/{org['login']})" for org in org_data]
+        orgs_to_add = ' | '.join(orgs)
+
+        # fetching starred repos data
+        starred_data = await self.fetch_data(user_data['starred_url'])
+
+        # forming blog link
+        if user_data['blog'].startswith("http"):  # blog link is clickable
+            blog = f"[Direct link]({user_data['blog']})"
+        elif user_data['blog']:  # blog exists but the link is not clickable
+            blog = f"[Direct link](https://{user_data['blog']})"
+        else:  # blog does not exist
+            blog = "No blog link available"
+
+        # building embed
```

Comment:
Unnecessary comment. This should be clear by looking at the next line alone. Also, inline comments should be used _sparingly_. They can clutter things up, so variable names and similar should make things self-explanatory, in most cases.

Quite a few of the comments feel redundant and I'd like to see many of them removed. Feel free to ask in #dev-contrib, and possibly ping me there, if you need help with deciding which comments to keep and which ones to remove.

Reply:
> Also, inline comments should be used sparingly.
Oh okay, I'll do this. It's just that I don't have a habit of commenting code and this is a first for me, I always thought that open source projects are "heavily" commented

Reply:
More often than not, that heavy commenting is docstrings. Inline comments aren't used that frequently.